### PR TITLE
Tanzu GemFire - Adding additional content

### PR DESCRIPTION
**Pull request by jomartin-999**
This commit adds content for:
- New Guides
- Additional blog articles about Tanzu GemFire.

It also fixes an issue with the sidebar na tree where blog articles were showing up in the guides section.

### CODE REVIEW

File: content/data/tanzu-gemfire/blog/apache-geode-apache-kafka.md [hunk 1]
```diff
@@ -0,0 +1,163 @@
...
+ start server --name=server1
+```
+
+![img](/images/data-blogs/tanzu-gemfire/apache-geode-apache-kafka/geode-kafka-gfsh.png)
+
+gfsh output after creating locator and server
+
+
+Create the region we want the data to end up in:
+create region --name=’Events’ --type=PARTITION
```
Comment (anthonyvetter) [inline content/data/tanzu-gemfire/blog/apache-geode-apache-kafka.md:48]:
```suggestion
Create the region we want the data to end up in:
```
create region --name=’Events’ --type=PARTITION
```
```

Comment (anthonyvetter) [inline content/data/tanzu-gemfire/blog/apache-geode-apache-kafka.md:48]: suggest wrapping in a code block if this is supposed to be a command the reader is intended to copy and paste out

File: content/data/tanzu-gemfire/blog/apache-geode-apache-kafka.md [hunk 2]
```diff
@@ -0,0 +1,163 @@
...
+### Create a topic named ‘test’
+
+```
+bin/kafka-topics.sh --create --bootstrap-server localhost:9092 --replication-factor 1 --partitions 1 --topic test
+```
+
+## Creating the Sink Connector
+We’ll now create a sink connector to flow data from Kafka to Geode. We will also configure the key and value converters. For the value converter we will explicitly use the JsonPdxConverter supplied by the geode-kafka-connector. For the key, we will use a String converter, where the example webapp will just populate with an arbitrary value.
+
+**Configure connect-geode-sink.properties**
```
Comment (anthonyvetter) [inline content/data/tanzu-gemfire/blog/apache-geode-apache-kafka.md:71]: This should be an H3 heading. Not just bolded

File: content/data/tanzu-gemfire/blog/apache-geode-apache-kafka.md [hunk 3]
```diff
@@ -0,0 +1,163 @@
...
+connector.class=GeodeKafkaSink
+tasks.max=1
+key.converter=org.apache.kafka.connect.storage.StringConverter
+value.converter=org.apache.geode.kafka.converter.JsonPdxConverter
+topic-to-regions=[test:Events]
+locators=localHost[10334]
+topics=test
+```
+
+**Modify config/connect-standalone.properties**
```
Comment (anthonyvetter) [inline content/data/tanzu-gemfire/blog/apache-geode-apache-kafka.md:85]: This should be an H3 heading. Not just bolded

File: content/data/tanzu-gemfire/blog/apache-geode-apache-kafka.md [hunk 4]
```diff
@@ -0,0 +1,163 @@
...
+## Ingest Data
+Hit the rest endpoint (note: if your topic wasn’t named test, you can simply change the topic name in the url)
+
+```
+curl localhost:8080/ingest/github/test
+```
+
+## Query the JSON data in Apache Geode!
+Revisit gfsh and set a gfsh environment variable so we can see values with large fields.
+set variable --name=APP_RESULT_VIEWER --value=external
```
Comment (anthonyvetter) [inline content/data/tanzu-gemfire/blog/apache-geode-apache-kafka.md:130]: This should probably be formatted as a code block

File: content/data/tanzu-gemfire/blog/apache-geode-apache-kafka.md [hunk 5]
```diff
@@ -0,0 +1,163 @@
...
+
+```
+query --query=”select * from /Events”
+```
+
+![img](/images/data-blogs/tanzu-gemfire/apache-geode-apache-kafka/geode-kafka-region-events.png)
+
+Events made it into the region and we are able to query for them!
+
+**Query nested fields**
```
Comment (anthonyvetter) [inline content/data/tanzu-gemfire/blog/apache-geode-apache-kafka.md:141]: This should be an H3 heading. Not just bolded

File: content/data/tanzu-gemfire/blog/apache-geode-metrics-wavefront.md [hunk 6]
```diff
@@ -0,0 +1,227 @@
...
+```
+
+## What Else?
+Also available is a [video walkthrough](https://www.youtube.com/watch?v=BDZh-FLkDTg)
+
+Join the [Apache Geode Community](https://geode.apache.org/community/)!
+
+[How to add more metrics to Apache Geodes implementation](https://cwiki.apache.org/confluence/display/GEODE/How+to+add+a+Meter)
+
+Create and share your own MetricsPublishingService on the [user list](http://apache%20geode%20community/)
```
Comment (anthonyvetter) [inline content/data/tanzu-gemfire/blog/apache-geode-metrics-wavefront.md:224]: This link doesn't work

File: content/data/tanzu-gemfire/blog/calculating-the-size-of-a-geode-region.md [hunk 7]
```diff
@@ -0,0 +1,206 @@
...
+  size = new SingleMemberRegionSize(region.getName(), regionSize, regionHistogram);
+ } catch (Exception e) {
+  cache.getLogger()
+   .warning("Caught the following exception attempting to calculate the size of region=" + region.getFullPath(), e);
+ }
+ return size;
+}
+```
+
+Accept or Reject Objects from Region Size
```
Comment (anthonyvetter) [inline content/data/tanzu-gemfire/blog/calculating-the-size-of-a-geode-region.md:83]: Should this be a heading? H2 or H3?

File: content/data/tanzu-gemfire/blog/command-region-pattern.md [hunk 8]
```diff
@@ -0,0 +1,163 @@
...
+- Juan Jose Ramos
+ 
+---
+
+## Introduction
+[Apache Geode](https://geode.apache.org/) is an in-memory data grid that provides real-time, consistent access to data-intensive applications throughout widely distributed cloud architectures. Its many features include fault-tolerance, high-availability, and WAN replication.
+
+The multi-site or WAN topology is used to connect distinct clusters, they act as one distributed system when they are coupled, and they act as independent systems when communication between sites fails.
+
+The *Command Pattern* is a behavioural design pattern in which an object is used to encapsulate all information needed to perform an action or trigger an event.
```
Comment (anthonyvetter) [inline content/data/tanzu-gemfire/blog/command-region-pattern.md:19]: behavioral

File: content/data/tanzu-gemfire/blog/command-region-pattern.md [hunk 9]
```diff
@@ -0,0 +1,163 @@
...
+![img](/images/data-blogs/tanzu-gemfire/command-region-pattern/diagrams/command-region-pattern-diagram.jpeg)
+
+1. The client application creates a DistributedCommand instance and executes a put operation through the PROXY region.
+2. The CacheWriter is invoked and the DistributedCommand is executed locally.
+3. The DistributedCommand is added to the gateway-sender queue for distribution to remote clusters.
+4. The DistributedCommand is distributed and received by a gateway-receiver on a remote cluster.
+5. The DistributedCommand reaches a remote server, where the CacheWriter is invoked and the DistributedCommand is executed locally.
+
+## Implementation
+* Create a *DistributedCommand* interface.
```
Comment (anthonyvetter) [inline content/data/tanzu-gemfire/blog/command-region-pattern.md:42]: This and all steps under this heading should be formatted as an ordered list. Numbered, as opposed to bulleted.

File: content/data/tanzu-gemfire/blog/apache-geode-metrics-wavefront.md (hunk 6 above)
Comment (jomartin-999) [reply content/data/tanzu-gemfire/blog/apache-geode-metrics-wavefront.md:224]: Thank you.  The link should be: https://geode.apache.org/community/

File: content/data/tanzu-gemfire/blog/command-region-pattern.md [hunk 10]
```diff
@@ -0,0 +1,163 @@
...
+[info 2020/04/10 13:57:24.341 IST <ServerConnection on port 40401 Thread 2> tid=0x51] Hello World from TestApplication1!.
+[info 2020/04/10 13:57:24.341 IST <ServerConnection on port 40401 Thread 2> tid=0x51] Executing distributed command HelloWorldCommand... Done!.
+-------------------------------------------------------------------
+[info 2020/04/10 13:57:24.873 IST <ServerConnection on port 5152 Thread 1> tid=0x42] Executing distributed command HelloWorldCommand...
+[info 2020/04/10 13:57:24.873 IST <ServerConnection on port 5152 Thread 1> tid=0x42] Hello World from TestApplication1!.
+[info 2020/04/10 13:57:24.873 IST <ServerConnection on port 5152 Thread 1> tid=0x42] Executing distributed command HelloWorldCommand... Done!.
+```
+
+## What’s Next?
+Check out [geode-command-region-pattern](https://github.com/jujoramos/geode-command-region-pattern) and play around with it, it allows you to test your commands in a distributed fashion using the [geode-dunit](https://cwiki.apache.org/confluence/display/GEODE/DistributedTest) module.
```
Comment (anthonyvetter) [inline content/data/tanzu-gemfire/blog/command-region-pattern.md:148]: geode-dunit link "page not found"

File: content/data/tanzu-gemfire/blog/converting-apache-geode-statistics-to-micrometer-meters.md [hunk 11]
```diff
@@ -0,0 +1,213 @@
...
+type: blog
+
+# Author(s) 
+team:
+- Barry Oglesby
+
+---
+
+## Introduction
+Apache Geode provides many Statistics for monitoring the Distributed System including:
```
Comment (anthonyvetter) [inline content/data/tanzu-gemfire/blog/converting-apache-geode-statistics-to-micrometer-meters.md:15]: Consider making Apach Geode a link to more information

File: content/data/tanzu-gemfire/blog/geode-distributed-sequences.md [hunk 12]
```diff
@@ -0,0 +1,92 @@
...
+If you are willing to allow gaps, continue reading further, if you don’t… well, this tool is not for you, don’t waste any more of your time :-).
+
+## The Solution
+Instead of embedding the sequence generation within the real-time processing, use the [Geode Sequence Generator](https://github.com/jujoramos/geode-sequence-generator) and invoke the service occasionally to retrieve a big enough set of counters to work on.
+Ideally, the service should be invoked during client startup and the batch requested should be big enough to avoid multiple calls to the service in real-time. The above alleviates the contention on the servers and maintains scalability, allowing several client applications to truly process batches of work in parallel using the set of assigned sequentially generated IDs.
+
+##How Does it Work?
```
Comment (anthonyvetter) [inline content/data/tanzu-gemfire/blog/geode-distributed-sequences.md:30]: Need the space after ## or it doesn't render the heading

File: content/data/tanzu-gemfire/blog/geode-distributed-sequences.md [hunk 13]
```diff
@@ -0,0 +1,92 @@
...
+To initialize the backing `Region`, the tool executes an initialization `Function` through the usage of `FunctionService.onServers`, allowing the user to choose exactly which servers will be used as the back end for the sequence generation (all by default).
+To generate or retrieve a batch of sequences, the tool executes a `Function` through `FunctionService.onRegion`, guaranteeing that only the server hosting the primary copy of the data will be reached.
+
+## Examples?
+All right, time to write some code, finally!.
+
+### Compile and Deploy
+The first step is to download the tool, build it and deploy it to a running Apache Geode cluster. We list the functions at the very end just to verify that they have been correctly registered through the gfsh deploy command.
+
+![img](/images/data-blogs/tanzu-gemfire/geode-distributed-sequences/geode-distributed-sequences-gfsh-deploy.png)
```
Comment (anthonyvetter) [inline content/data/tanzu-gemfire/blog/geode-distributed-sequences.md:42]: In the future, we prefer any commands that a user is intended to run be formatted as code blocks as opposed to screen shot images. If you want to include command output as a means of verification, that can also be a code block or an image, but preferably is limited to one command at a time.

File: content/data/tanzu-gemfire/blog/improving-the-performance-of-apache-geode-persistence-recovery.md [hunk 14]
```diff
@@ -0,0 +1,88 @@
...
+
+# Author(s)
+team: 
+- Jianxia Chen
+ 
+---
+
+## Introduction
+
+Apache Geode is a data management platform that provides real-time, consistent access to data-intensive applications throughout widely distributed cloud architectures. Geode pools memory, CPU, network resources, and optionally local disk across multiple processes to manage application objects and behavior. It uses dynamic replication and data partitioning techniques to implement high availability, improved performance, scalability, and fault tolerance. In addition to being a distributed data container, Apache Geode is an in-memory data management system that provides reliable asynchronous event notifications and guaranteed message delivery.
```
Comment (anthonyvetter) [inline content/data/tanzu-gemfire/blog/improving-the-performance-of-apache-geode-persistence-recovery.md:17]: Consider linking Apache Geode to additional information

File: content/data/tanzu-gemfire/blog/improving-the-performance-of-apache-geode-persistence-recovery.md [hunk 15]
```diff
@@ -0,0 +1,88 @@
...
+
+The article assumes you have a basic understanding of Apache Geode. You can refer to the [quick start](https://geode.apache.org/docs/guide/19/getting_started/book_intro.html) for an overview of the system.
```
Comment (anthonyvetter) [inline content/data/tanzu-gemfire/blog/improving-the-performance-of-apache-geode-persistence-recovery.md:19]: The article --> This article ?

File: content/data/tanzu-gemfire/blog/improving-the-performance-of-apache-geode-persistence-recovery.md [hunk 16]
```diff
@@ -0,0 +1,88 @@
...
+
+With the help of the warning messages, and the thread stacks, we have identified that in the source code, the synchronization of `HashMap` has caused the persistence recovery to slow down significantly. The thread synchronization on server1 has affected not only server1 itself, but also the progress of server2. For log analysis details, please refer to [GEODE-7945](https://issues.apache.org/jira/browse/GEODE-7945) and its pull request.
+
+Once we understand the root cause, the solution becomes clear. We’d better replace `HashMap` with `ConcurrentHashMap`, to remove unnecessary synchronization among the threads. With `ConcurrentHashMap`, the persistence recovery time is reduced by 30% for recovering the same amount of persistent data during Geode cluster restart.
+
+## Parallel Disk stores Recovery
```
Comment (anthonyvetter) [inline content/data/tanzu-gemfire/blog/improving-the-performance-of-apache-geode-persistence-recovery.md:70]: Stores should be capitalized

File: content/data/tanzu-gemfire/blog/calculating-the-size-of-a-geode-region.md (hunk 7 above)
Comment (jomartin-999) [reply content/data/tanzu-gemfire/blog/calculating-the-size-of-a-geode-region.md:83]: H3

File: content/data/tanzu-gemfire/blog/command-region-pattern.md (hunk 10 above)
Comment (jomartin-999) [reply content/data/tanzu-gemfire/blog/command-region-pattern.md:148]: Sorry.  Page was renamed.  Link should be: https://cwiki.apache.org/confluence/display/GEODE/About+Distributed+Testing

File: content/data/tanzu-gemfire/blog/converting-apache-geode-statistics-to-micrometer-meters.md [hunk 17]
```diff
@@ -0,0 +1,213 @@
...
+
+In addition, set the properties below in the application.properties file. The last two are optional, but I was seeing intermittent timeouts, so I increased the defaults.
+
+| Property          | Description     | Example     |
+| ----------------- | ---------------- | --------------- | 
+| management.metrics.export.wavefront.api-token | API token | Generate in your account |
+| management.metrics.export.wavefront.uri | Wavefront URI | https://vmware.wavefront.com |
+| management.metrics.export.wavefront.source | Application identifier | my.test.application |
+| management.metrics.export.wavefront.connect-timeout | Connection timeout | 10s |
+| management.metrics.export.wavefront.read-timeout | Read timeout | 60s |
```
[the remark this answers is not in the document]
Comment (jomartin-999) [reply content/data/tanzu-gemfire/blog/converting-apache-geode-statistics-to-micrometer-meters.md:193]: I didn't write the original blog, so I don't feel comfortable removing text.  I tried putting it in a code-block, but that seemed to make it worse.  A different format may work.

File: content/data/tanzu-gemfire/blog/improving-the-performance-of-apache-geode-persistence-recovery.md (hunk 15 above)
Comment (jomartin-999) [reply content/data/tanzu-gemfire/blog/improving-the-performance-of-apache-geode-persistence-recovery.md:19]: Yes, good catch.  Thank you.

File: content/data/tanzu-gemfire/guides/cache-aside-pattern-sbdg.md [hunk 18]
```diff
@@ -0,0 +1,452 @@
...
+ **Maven**
+```
+mvn spring-boot:run
+``` 
+
+When the app is running, open a browser and go to <http://localhost:8080>.  You should see this
+
+![img](/images/guides/data/tanzu-gemfire/cache-aside-sbdg/cache-aside-pattern-1.png)
+
+&nbsp;
```
Comment (BrianMMcClain) [inline content/data/tanzu-gemfire/guides/cache-aside-pattern-sbdg.md:257]: We should probably remove these

File: content/data/tanzu-gemfire/guides/get-started-tgf4k8s-sbdg.md [hunk 19]
```diff
@@ -0,0 +1,359 @@
...
+ 
+
+## Create A Tanzu GemFire Cluster
+
+1. Verify that you are in the Kubernetes cluster you want to use for Tanzu GemFire
+
+    ```
+    kubectl config current-context
+    ```
+   &nbsp;
```
Comment (BrianMMcClain) [inline content/data/tanzu-gemfire/guides/get-started-tgf4k8s-sbdg.md:32]: Likewise, these should probably be removed from here as well

File: content/data/tanzu-gemfire/blog/pluggable-oql-method-auth.md [hunk 20]
```diff
@@ -0,0 +1,121 @@
...
+### *UnrestrictedMethodAuthorizer*
+The “I want it all” one, it allows every method invocation except for the ones that are already flagged as “dangerous” by default, that is, those methods that can affect the internals of the cache and its regions and, of course, [Java reflection](https://docs.oracle.com/javase/8/docs/technotes/guides/reflection/index.html). Use this one for secured clusters where only trusted applications have access to the query engine, or whenever all entries within the cache are immutable.
+
+To configure this authorizer in your cluster, execute the following command
+```
+gfsh alter query-service --method-authorizer=org.apache.geode.cache.query.security.UnrestrictedMethodAuthorizer 
+```
+
+### *JavaBeanAccessorMethodAuthorizer*
+The “play by the book” one, allowing only those methods from configured packages that follow the [JavaBean Specification 1.01](https://download.oracle.com/otndocs/jcp/7224-javabeans-1.01-fr-spec-oth-JSpec/); that is, methods that start with `get` or `is` followed by the attribute name. Use this one only when you’re 100% sure that all the developers have followed the mentioned specification
```
Comment (anthonyvetter) [inline content/data/tanzu-gemfire/blog/pluggable-oql-method-auth.md:55]: needs a period at the end of the paragraph

File: content/data/tanzu-gemfire/blog/pluggable-oql-method-auth.md [hunk 21]
```diff
@@ -0,0 +1,121 @@
...
+The [Apache Geode community](https://geode.apache.org/community/) learns from previous errors and iterates to improve the end-user experience, that’s why the decision was made to change this “all or nothing approach” and provide users the ability to either choose from some general targeted use case out of box authorization implementations or create their own.
+
+## Out of the Box Implementations
+
+Based on feedback from users, the [Apache Geode community](https://geode.apache.org/community/) tried to cover the most common use cases and scenarios when designing and implementing these authorizers; the full list with a brief description for each one is shown below.
+
+### *RestrictedMethodAuthorizer*
+The default “please secure it all” one, it denies everything except for the hard-coded list of known safe methods. Use this if you liked the approach introduced in [Apache Geode 1.3.0](https://archive.apache.org/dist/geode/1.3.0/).
+
+To configure this authorizer in your cluster, just execute the following command
```
Comment (anthonyvetter) [inline content/data/tanzu-gemfire/blog/pluggable-oql-method-auth.md:41]: Needs a period or a colon at the end

**Comment by BrianMMcClain** [comment]
Hi @jomartin-999, @anthonyvetter and I finished up our review on here, once the changes are made and any merge conflicts are resolved, we can get it merged.

**Review by BrianMMcClain — CHANGES_REQUESTED** [review]

File: content/data/tanzu-gemfire/blog/geode-distributed-sequences.md (hunk 13 above)
Comment (jomartin-999) [reply content/data/tanzu-gemfire/blog/geode-distributed-sequences.md:42]: Yes, completely agree.  Moving forward this is something we will continue to communicate throughout the team.

**Review by BrianMMcClain — APPROVED** [review]

**Pull request merged**